### DISCUSSION TITLE
Artifacthub Terraform Provider

### DISCUSSION THREAD

**Discussion by cgetzen** [artifacthub/hub#3984] [Ideas]
**Is your feature request related to a problem? Please describe.**
We would like to automate the management of repositories inside of artifacthubs.

**Describe the solution you'd like**
- An initial terraform provider with authentication to a custom artifacthub
- An `artifacthub_repository` (or `artifacthub_repository_helm`) resource 

**Describe alternatives you've considered**
- Using a terraform local-exec hook to perform API requests in artifacthub (specifically `PUT /repositories/org/{orgName}/{repoName}`)
- Configuring these repos through helm, similar to `loadSampleData`

**Additional context**
N/A

**Comment by tegioz** [comment]
Hi @cgetzen 👋

This is an interesting idea, but I don't think it is within the scope of the project at the moment. But of course you are welcome to implement this as an external add-on using the Artifact Hub API as you mentioned 🙂

**Comment by tegioz** [comment]
Will close this issue for now, please feel free to reopen if needed 🙂

**Comment by tegioz** [comment]
I've converted the issue to a discussion and moved it to the `ideas` category @cgetzen.
